app/javascript/react/screens/App/Settings/Settings.js

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { reduxForm, Field } from 'redux-form';
+import { Breadcrumb, Form, Grid, Button, Spinner } from 'patternfly-react';
+import Toolbar from '../../../config/Toolbar';
+import NumberInput from '../common/forms/NumberInput';
+
+class Settings extends React.Component {
+  componentDidMount() {
+    const { fetchServersAction, fetchServersUrl, fetchSettingsAction, fetchSettingsUrl } = this.props;
+    fetchServersAction(fetchServersUrl);
+    fetchSettingsAction(fetchSettingsUrl);
+  }
+
+  onApplyClick = () => {
+    const { patchSettingsAction, servers, settingsForm } = this.props;
+    patchSettingsAction(servers, settingsForm.values);
+  };
+
+  render() {
+    const { isFetchingServers, isFetchingSettings, isSavingSettings, savedSettings, settingsForm } = this.props;
+
+    const toolbarContent = (
+      <Toolbar>
+        <Breadcrumb.Item href="/dashboard/maintab?tab=compute">{__('Compute')}</Breadcrumb.Item>
+        <Breadcrumb.Item href="/migration">{__('Migration')}</Breadcrumb.Item>
+        <Breadcrumb.Item active>{__('Migration Settings')}</Breadcrumb.Item>
+      </Toolbar>
+    );
+
+    const hasUnsavedChanges =
+      settingsForm &&
+      settingsForm.values &&
+      Object.keys(savedSettings).some(key => savedSettings[key] !== settingsForm.values[key]);
+
+    const settingsContent = (
+      <Spinner loading={isFetchingServers || isFetchingSettings} style={{ marginTop: 15 }}>
+        <div className="migration-settings">
+          <h2>{__('Concurrent Migrations')}</h2>
+          <Form horizontal>
+            <Form.FormGroup>
+              <Grid.Col componentClass={Form.ControlLabel} sm={5}>
+                {__('Maximum concurrent migrations per conversion host')}
+              </Grid.Col>
+              <Grid.Col sm={7}>
+                <div style={{ width: 100 }}>
+                  <Field
+                    id="max_concurrent_tasks_per_host"
+                    name="max_concurrent_tasks_per_host"
+                    component={NumberInput}
+                    normalize={NumberInput.normalizeStringToInt}
```

Comment:
I'm probably just missing something, but can you clarify where this is being used?  Also, I think this prop is being passed down into the `NumberInput` component, so if `normalizeStringToInt` is an instance property of the class, do we need it here?

Reply:
nvm, just checked.  This is a special `redux-form` thing.  Nothing to see here